cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/store/DefaultMetadataStore.java

```diff
@@ -148,16 +157,39 @@ public void setProperties(final MetadataScope scope, final MetadataEntity metada
                  new MetadataRecordV2(metadataEntity, scope, propDeletions.build(), EMPTY_TAGS));
   }
 
+  private Boolean hasEntityInV1(MetadataScope scope, MetadataEntity metadataEntity) {
+    return execute(mds -> !mds.getProperties(metadataEntity).isEmpty() ||
+      !mds.getTags(metadataEntity).isEmpty(), scope, DATASET.V1);
+  }
+
+  private void updateInMemoryState() {
```

Comment:
Call this `boolean isMigrationCompleted()`. How it uses the field to store states is irrelevant.